pkg/coap/coap.go

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package coap
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/absmach/mproxy"
+	"github.com/absmach/mproxy/pkg/session"
+	"github.com/plgd-dev/go-coap/v3/dtls"
+	"github.com/plgd-dev/go-coap/v3/message"
+	"github.com/plgd-dev/go-coap/v3/message/codes"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	"github.com/plgd-dev/go-coap/v3/mux"
+	"github.com/plgd-dev/go-coap/v3/net"
+	"github.com/plgd-dev/go-coap/v3/options"
+	"github.com/plgd-dev/go-coap/v3/udp"
+)
+
+type Proxy struct {
+	config mproxy.Config
+	event  session.Handler
+	logger *slog.Logger
+}
+
+func New(config mproxy.Config, handler session.Handler, logger *slog.Logger) *Proxy {
```

Comment:
New is a go keyword. Let's rename it to something like

```suggestion
func NewProxy(config mproxy.Config, handler session.Handler, logger *slog.Logger) *Proxy {
```